spec/models/spree/line_item_spec.rb

```diff
@@ -132,6 +132,59 @@ module Spree
       end
     end
 
+    describe "reducing stock levels on order completion" do
+      context "when the item is on_demand" do
+        let!(:hub) { create(:distributor_enterprise) }
+        let(:bill_address) { create(:address) }
+        let!(:variant_on_demand) { create(:variant, on_demand: true, on_hand: 1) }
+        let!(:order) {
+          create(:order_with_totals_and_distribution,
+                 distributor: hub,
+                 bill_address: bill_address,
+                 ship_address: bill_address)
+        }
+        let!(:shipping_method) { create(:shipping_method, distributors: [hub]) }
+        let!(:line_item) { create(:line_item, variant: variant_on_demand, quantity: 10, order: order) }
+
+        before do
+          order.update_totals
+          order.payments << create(:payment, amount: order.total)
+          until order.completed? do break unless order.next! end
+          order.payment_state = 'paid'
+          order.select_shipping_method(shipping_method.id)
+          order.shipment.update!(order)
+        end
+
+        it "creates a shipment without backordered items" do
+          expect(order.shipment.manifest.first.quantity).to eq 10
+          expect(order.shipment.manifest.first.states).to eq 'on_hand' => 10
+          expect(order.shipment.manifest.first.variant).to eq line_item.variant
+        end
+
+        it "does not reduce the variant's stock level" do
+          expect(variant_on_demand.reload.on_hand).to eq 1
+        end
+
+        it "does not mark inventory units as backorderd" do
+          backordered_units = order.shipments.first.inventory_units.any?(&:backordered?)
+          expect(backordered_units).to be false
+        end
+
+        it "does not mark the shipment as backorderd" do
+          expect(order.shipments.first.backordered?).to be false
+        end
+
+        it "allows the order to be shipped" do
+          expect(order.ready_to_ship?).to be true
+        end
+
+        it "does not change stock levels when cancelled" do
+          order.cancel!
+          expect(variant_on_demand.reload.on_hand).to eq 1
+        end
+      end
+    end
```

Comment:
IMO this belongs to a layer above, controller tests. This file contains line items unit tests but the examples that we added not even reference a line item.